frontend/talentsearch/src/js/components/governmentInfoForm/GovernmentInfoForm.tsx

```diff
@@ -0,0 +1,431 @@
+    return idConnect;
+  };
+
+  // submitting the form component values back out to graphQL, after smoothing form-values to appropriate type, then return to /profile
+  const [, executeMutation] = useUpdateGovAsUserMutation();
+  const handleUpdateUser = (id: string, data: UpdateUserAsUserInput) =>
+    executeMutation({
+      id,
+      user: data,
+    }).then((result) => {
+      if (result.data?.updateUserAsUser) {
+        return result.data.updateUserAsUser;
+      }
+      return Promise.reject(result.error);
+    });
+  const formValuesToSubmitData = (
+    values: FormValues,
+  ): UpdateUserAsUserInput => ({
+    isGovEmployee: values.govEmployeeYesNo === "yes",
+    interestedInLaterOrSecondment: values.lateralDeployBool,
+    currentClassification: {
+      connect: classificationFormToId(
+        values.currentClassificationGroup,
+        values.currentClassificationLevel,
+      ),
+    },
+  });
+  const id = holder ? holder.id : "";
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    await handleUpdateUser(id, formValuesToSubmitData(data))
+      .then(() => {
+        navigate(paths.profile());
+        toast.success(
+          intl.formatMessage({
+            defaultMessage: "User updated successfully!",
+            description:
+              "Message displayed to user after user is updated successfully.",
+          }),
+        );
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Error: updating user failed",
+            description:
+              "Message displayed to user after user fails to get updated.",
+          }),
+        );
+      });
+  };
+
+  if (fetchingLookupData || fetchingMe) {
+    return (
+      <p>
+        {intl.formatMessage({
+          defaultMessage: "Loading...",
+          description: "Loading message",
+        })}
+      </p>
+    );
+  }
+
+  return (
+    <div>
+      <ProfileFormWrapper
+        description={intl.formatMessage({
+          defaultMessage:
+            "Please indicate if you are currently an employee in the Government of Canada.",
+          description:
+            "Description blurb for Profile Form Wrapper in the Government Information Form",
+        })}
+        title={intl.formatMessage({
+          defaultMessage: "Government Information",
+          description:
+            "Title for Profile Form Wrapper in Government Information Form",
+        })}
+        crumbs={[
+          {
+            title: intl.formatMessage({
+              defaultMessage: "Government Information",
+              description:
+                "Display Text for Government Information Form Page Link",
+            }),
+          },
+        ]}
+      >
+        <FormProvider {...methods}>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <GovernmentInfoForm
+              classifications={classifications}
+              storedValues={previousData}
```

Comment:
It doesn't look like GovernmentInfoForm is actually using previousData. And that's fine. Really, we should be passing previousData into useForm as [defaultValues](https://react-hook-form.com/api/useform).

Actually... this might be a bit tricky? useForm only uses defaultValues on first render. And the previousValues are not defined at first because they come from the api... Its possible you'll need to use `useEffect` to reset the form values the first time previousValues are populated? 

Also, you might be able to get around all this by using `<BasicForm>` instead of the useForm hook and FormProvider. That way, useForm will never be called until the api calls are done fetching.

Reply:
Yeah, trying to pre-fill values horribly broke things for me, it didn't play nice with either conditional rendering or how the Select options are made